src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -200,6 +187,15 @@ private void emitSegmentGetter(String javaName, String offsetField, String sizeF
             """);
     }
 
+    private void emitAsSlice() {
+        appendIndentedLines(STR."""
+
+            public static MemorySegment asSlice(MemorySegment ptr, long index) {
```

Comment:
this might clash, right? Perhaps we can detect clashes locally, e.g. inside StructBuilder?